### PR TITLE
Transcripts - Support json format

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
This supports JSON transcript format.

## Testing Instructions
1. Prioritise `TranscriptFormat.JSON_PODCAST_INDEX` in `TranscriptsManagerImpl.supportedFormats` at line 22
2. Install the app 
3. Play [this episode](https://pca.st/episode/cadc4a0f-97c3-47c6-9e7d-16452a2cf18a) from `Chaosradio`
4. Open transcript view
5. Notice that JSON format transcript is loaded

## Screenshots or Screencast 

Loaded from [transcript content](https://chaosradio.de/cr291-vorratsdatenspeicherung?podlove_transcript=json_podcastindex) 

## Checklist
- [x] ~~If this is a user-facing change, I have added an entry in CHANGELOG.md~~
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] ~~All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`~~
- [x] ~~Any jetpack compose components I added or changed are covered by compose previews~~
- [x] ~~I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.~~
 
#### I have tested any UI changes...

- [x] ~~with different themes~~
- [x] ~~with a landscape orientation~~
- [x] ~~with the device set to have a large display and font size~~
- [x] ~~for accessibility with TalkBack~~

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>TranscriptSegments</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptJsonParser.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    fun parse(jsonString: String): List<TranscriptCue> {
+        val jsonAdapter: JsonAdapter<TranscriptSegments> =
+            moshi.adapter(TranscriptSegments::class.java)
+        val transcriptJson =
+            jsonAdapter.fromJson(jsonString) ?: throw IllegalArgumentException("Invalid JSON")
+
+        return transcriptJson.segments
+    }
+}
```
Comment (MiSikora) [inline modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptJsonParser.kt:33]: Can you change that to use DI? Inject `Moshi` that is already available in our graph either to the view model or to a wrapper if you think it fits better.

**Review by MiSikora — COMMENTED** [review]
> Prioritise `TranscriptFormat.JSON_PODCAST_INDEX` in `TranscriptsManagerImpl.supportedFormats` at line 22

I'm a bit confused by this requirement. Do you plan to add detection of the best format to display in a different PR?

**Comment by ashiagr** [comment]
> I'm a bit confused by this requirement. Do you plan to add detection of the best format to display in a different PR?

Yes, it will be handled in a different PR as part of this subtask in [GitHub Project](https://github.com/Automattic/pocket-casts-android/issues/2398):
- Fallback to alternative formats if main format fails

**Review by MiSikora — APPROVED** [review]

File: modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptJsonParser.kt (hunk 1 above)
Comment (ashiagr) [reply modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/transcripts/TranscriptJsonParser.kt]: Injected in 58786ba

**Pull request merged**